source/utils/paragraphHelper.py

```diff
@@ -0,0 +1,252 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited, Rob Meredith
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import api
+import speech
+import controlTypes
+import textInfos
+import tones
+import config
+from NVDAObjects.window.winword import WordDocumentTextInfo
+from NVDAObjects.window.winword import BrowseModeWordDocumentTextInfo
+from NVDAObjects.UIA import UIATextInfo
+from displayModel import EditableTextDisplayModelTextInfo
+
+MAX_LINES = 250  # give up after searching this many lines
+
+
+def nextParagraphStyle() -> config.featureFlag.FeatureFlag:
+	from config.featureFlagEnums import ParagraphNavigationFlag
+	flag: config.featureFlag.FeatureFlag = config.conf["documentNavigation"]["paragraphStyle"]
+	numStyles = len(ParagraphNavigationFlag.__members__)
+	newEnumVal = flag.calculated().value + 1
+	if newEnumVal > numStyles:
+		newEnumVal = 2  # wrap around, skip DEFAULT
+	return ParagraphNavigationFlag(newEnumVal)
+
+
+def getTextInfoAtCaret() -> textInfos.TextInfo:
+	# returns None if not editable text or document
+	ti = None
+	focus = api.getFocusObject()
+	if (focus.role == controlTypes.Role.EDITABLETEXT) or (focus.role == controlTypes.Role.DOCUMENT):
+		try:
+			ti = focus.makeTextInfo(textInfos.POSITION_CARET)
+		except (NotImplementedError, RuntimeError):
+			pass
+	return ti
+
+
+def isAcceptableTextInfo(ti: textInfos.TextInfo) -> bool:
```

Comment:
this may be worth also marking as internal API

```suggestion
def _isAcceptableTextInfo(ti: textInfos.TextInfo) -> bool:
```